requirements.txt

```diff
@@ -40,3 +40,4 @@ whatthepatch==1.0.0
 wordcloud==1.7.0
 wrapt==1.12.1
 zipp==3.1.0
+cachetools=4.1.1
```

Comment:
should be `==`

Reply:
Oh boy...

Reply:
also the dask + tlz dependency is missing in the requirements

Reply:
Hmm, I don't have tlz in my environment at all?

Reply:
The lib is called `toolz`. After I install all the requirements from the `requirements.txt` I am getting this error:
```
python bohr/pipeline/apply_heuristics.py --rows-train 50000 --rows-test 5000 --save-heuristics-matrix-train-to generated/heuristic_matrix_train.pkl --save-heuristics-matrix-test-to generated/heuristic_matrix_test.pkl --save-metrics-to heuristic_metrics.json
Traceback (most recent call last):
  File "/Users/hlib/.pyenv/versions/bohr/lib/python3.8/site-packages/dask/dataframe/__init__.py", line 2, in <module>
    from .core import (
  File "/Users/hlib/.pyenv/versions/bohr/lib/python3.8/site-packages/dask/dataframe/core.py", line 18, in <module>
    from tlz import merge, first, unique, partition_all, remove
ModuleNotFoundError: No module named 'tlz'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "bohr/pipeline/apply_heuristics.py", line 12, in <module>
    import dask.dataframe as dd
  File "/Users/hlib/.pyenv/versions/bohr/lib/python3.8/site-packages/dask/dataframe/__init__.py", line 57, in <module>
    raise ImportError(msg) from e
ImportError: Dask dataframe requirements are not installed.

Please either conda or pip install as follows:

  conda install dask                     # either conda install
  python -m pip install "dask[dataframe]" --upgrade  # or python -m pip install
```

Anyway, I ran `pip install -r requirements.txt`, installed what was missing and did `pip freeze > requirements.txt`. Now my `requirements.txt` looks like this:
```
absl-py==0.9.0
cachetools==4.1.1
certifi==2020.6.20
chardet==3.0.4
click==7.1.2
cloudpickle==1.6.0
cycler==0.10.0
Cython==0.29.21
dask==2.27.0
decorator==4.4.2
Deprecated==1.2.10
distributed==2.27.0
fsspec==0.8.2
future==0.18.2
grpcio==1.30.0
HeapDict==1.0.1
idna==2.10
importlib-metadata==1.7.0
joblib==0.16.0
kiwisolver==1.2.0
locket==0.2.0
Markdown==3.2.2
matplotlib==3.2.2
msgpack==1.0.0
munkres==1.1.2
networkx==2.3
nltk==3.5
numpy==1.19.0
pandas==0.25.3
partd==1.1.0
Pillow==7.2.0
protobuf==3.12.2
psutil==5.7.2
PyGithub==1.51
PyJWT==1.7.1
pyparsing==2.4.7
python-dateutil==2.8.1
pytz==2020.1
PyYAML==5.3.1
regex==2020.6.8
requests==2.24.0
scikit-learn==0.21.3
scipy==1.5.1
six==1.15.0
snorkel==0.9.5
sortedcontainers==2.2.2
tblib==1.7.0
tensorboard==1.15.0
toolz==0.11.0
torch==1.5.1
tornado==6.0.4
tqdm==4.47.0
urllib3==1.25.9
Werkzeug==1.0.1
whatthepatch==1.0.0
wordcloud==1.7.0
wrapt==1.12.1
zict==2.0.0
zipp==3.1.0
```